vscode-extensions/playwright-on-codespaces-vscode-extension/src/extension.ts

```diff
@@ -0,0 +1,284 @@
+        Buffer.from('This is a test file created by the Playwright on Codespaces extension.\n', 'utf8')
+      );
+      terminal.writeLine(`Test file written to temp directory.`);
+    }
+  } catch (error) {
+    terminal.writeError(`Error writing extension installed file: ${error}`);
+    return;
+  }
+}
+
+export async function activate(context: vscode.ExtensionContext): Promise<void> {
+  // Setup Logging Terminal
+  const outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel(EXTENSION_DISPLAY_NAME);
+  outputChannel.appendLine(`${EXTENSION_DISPLAY_NAME} Extension output channel initialized.`);
+
+  // Create terminal adapter for PlaywrightTunnel
+  const terminalProvider: ITerminalProvider = new VScodeOutputChannelTerminalProvider(outputChannel, {
+    debugEnabled: true,
+    verboseEnabled: true
+  });
+
+  const terminal: ITerminal = new Terminal(terminalProvider);
+
+  await writeExtensionInstalledFile(terminal);
+
+  // Create status bar item
+  const statusBarItem: vscode.StatusBarItem = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Right,
+    100
+  );
+
+  statusBarItem.command = COMMAND_SHOW_MENU;
+  let currentStatus: TunnelStatus = 'stopped';
+
+  function updateStatusBar(status: TunnelStatus): void {
+    currentStatus = status;
+    switch (status) {
+      case 'stopped':
+        statusBarItem.text = '$(debug-stop) Playwright Tunnel';
+        statusBarItem.tooltip = 'Playwright Tunnel: Stopped - Click for options';
+        statusBarItem.backgroundColor = undefined;
+        break;
+      case 'waiting-for-connection':
+        statusBarItem.text = '$(sync~spin) Playwright Tunnel';
+        statusBarItem.tooltip = 'Playwright Tunnel: Waiting for connection...';
+        statusBarItem.backgroundColor = new vscode.ThemeColor('statusBarItem.warningBackground');
+        break;
+      case 'setting-up-browser-server':
+        statusBarItem.text = '$(loading~spin) Playwright Tunnel';
+        statusBarItem.tooltip = 'Playwright Tunnel: Setting up browser server...';
+        statusBarItem.backgroundColor = new vscode.ThemeColor('statusBarItem.warningBackground');
+        break;
+      case 'browser-server-running':
+        statusBarItem.text = '$(check) Playwright Tunnel';
+        statusBarItem.tooltip = 'Playwright Tunnel: Running';
+        statusBarItem.backgroundColor = new vscode.ThemeColor('statusBarItem.prominentBackground');
+        break;
+      case 'error':
+        statusBarItem.text = '$(error) Playwright Tunnel';
+        statusBarItem.tooltip = 'Playwright Tunnel: Error - Click for options';
+        statusBarItem.backgroundColor = new vscode.ThemeColor('statusBarItem.errorBackground');
+        break;
+    }
+  }
+
+  // Initialize status bar
+  updateStatusBar('stopped');
+  statusBarItem.show();
+
+  // Tunnel instance
+  let tunnel: PlaywrightTunnel | undefined;
+
+  function getTmpPath(): string {
+    return path.join(os.tmpdir(), 'playwright-browser-tunnel');
```

Comment:
```suggestion
    return `${os.tmpdir()}/playwright-browser-tunnel`;
```

Should the temp folder path be configurable?